doc/reference/trd-appid.md

```diff
@@ -0,0 +1,790 @@
+reserved for footers.
+
+This is the format of a Program Header:
+
+```
+0             2             4             6             8
++-------------+-------------+---------------------------+
+| Type (9)    | Length (16) | init_fn_offset            |
++-------------+-------------+---------------------------+
+| protected_size            | min_ram_size              |
++---------------------------+---------------------------+
+| binary_end_offset         | version                   |
++---------------------------+---------------------------+
+```
+
+It is represented in the Tock kernel with this Rust structure:
+
+```rust
+pub struct TbfHeaderV2Program {
+    init_fn_offset: u32,
+    protected_size: u32,
+    minimum_ram_size: u32,
+    binary_end_offset: u32,
+    version: u32,
+}
+```
+
+A TBF object MUST NOT have more than one Program Header. If a TBF
+Object has both a Program Header and a Main Header, the kernel's
+policy decides which is used. For example, older kernels that do not
+understand a Program Header may use the Main Header, while newer
+kernels may choose the Program Header.
+
+5.2 Credentials Footer
+-------------------------------
+
+To support credentials, the Tock Binary Format has a
+`TbfFooterV2Credentials` TLV. This TLV is variable length and has two
+fields, a 32-bit value specifying the `format` of the credentials and
+a variable length `data` field. The `format` field defines the format
+and size of the `data` field. Each value of the `format` field except
+`Padding` MUST have a fixed data size and format. This is the format
+of a Credentials Footer:
+
+```
+0             2             4                           8
++-------------+-------------+---------------------------+
+| Type (128)  | Length      | format                    |
++-------------+-------------+---------------------------+
+| data                      |
++-------------+--------...--+
+```
+
+It is represented in the Tock kernel with this structure:
+
+```rust
+pub struct TbfFooterV2Credentials {
+    format: TbfFooterV2CredentialsType,
+    data: &[u8],
+}
+```
+
+Currently supported values of `format` are:
+
+```rust
+pub enum TbfFooterV2CredentialsType {
+    Padding = 0,
+    CleartextID = 1,
+    Rsa3072Key = 2,
+    Rsa4096Key = 3,
+    Rsa3072KeyWithID = 4,
+    Rsa4096KeyWithID = 5,
+	SHA256 = 6,
+	SHA384 = 7,
+	SHA512 = 8,
+}
+```
+
+The `Padding` type has a variable length. This credentials type is
+used to reserve space for future credentials or pad their placement.
+
+The `CleartextID` type has a data length of 8 bytes. It contains a
+64-bit number in big-endian format representing an application
+identifier.
+
+The `Rsa3072Key` type has a data of length of 768 bytes. It contains a
+public 3072-bit RSA key (384 bytes), followed by a 384-byte PKCS#1
+v1.5 signature using SHA512 (`CKM_SHA512_RSA_PKCS`). It does not
+contain a public exponnent: the Process Checker is responsible for
+storing the public exponent for any key it recognizes.
```

Comment:
```suggestion
v1.5 signature using SHA512 (`CKM_SHA512_RSA_PKCS`). It does not
contain a public exponent: the Process Checker is responsible for
storing the public exponent for any key it recognizes.
```

Also, is this explained somewhere, or is this common practice, or can we record the intent here? Mostly because I'm curious.

Reply:
The idea here is that the public key is the identifier of what (key, exponent) pair to use. Generally there's a small set of exponents that are used in practice. 

I would not necessarily see this particular specification as final -- we want to check with OT on what a good "standard" format would be.

Reply:
> I would not necessarily see this particular specification as final -- we want to check with OT on what a good "standard" format would be.

Makes sense, if we decide to save the 512 bytes we could change it or add a new credential type.